llvm/lib/CodeGen/InterleavedAccessPass.cpp

```diff
@@ -698,7 +701,7 @@ bool InterleavedAccessImpl::lowerDeinterleaveIntrinsic(
 
     // Since lowerInterleaveLoad expects Shuffles and LoadInst, use special
     // TLI function to emit target-specific interleaved instruction.
-    if (!TLI->lowerInterleavedVPLoad(VPLoad, Mask, DeinterleaveValues))
+    if (!TLI->lowerInterleavedVPLoad(VPLoad, Mask, DeinterleaveValues, Factor))
```

Comment:
Should we just plumb through `LastFactor->getType()` to save TLIs from having to do find_if on the DeinterleaveValues each time?